ModRemote.rbxs

```diff
@@ -1,423 +1,391 @@
+-- @Author Vorlias
+-- Edited by Narrev
+
 --[[
-	ModRemote v3.18
+	ModRemote v4.00
 		ModuleScript for handling networking via client/server
 		
 	Documentation for this ModuleScript can be found at
 		https://github.com/VoidKnight/ROBLOX-RemoteModule/tree/master/Version-3.x
 ]]
--- Main variables
-local replicated = game:GetService("ReplicatedStorage");
-local server = game:FindService("NetworkServer");
-local remoteStorage = replicated;
-
-local functionStorage = remoteStorage:FindFirstChild("RemoteFunctions") or Instance.new("Folder",remoteStorage);
-functionStorage.Name = "RemoteFunctions";
-
-local eventStorage = remoteStorage:FindFirstChild("RemoteEvents") or Instance.new("Folder",remoteStorage);
-eventStorage.Name = "RemoteEvents";
-
-local filtering = workspace.FilteringEnabled;
-local localServer = (game.JobId == '');
-
-local client = not server;
-local remote = {
-	Events = {};
-	Functions = {};
-	event = {};
-	func = {};
-	internal = {};
-	Version = 3.18;
-};
-
--- This warning will only show on the server
-if (not filtering and server and not remote.HideFilteringWarning) then
-	error("[ModRemote] ModRemote 3.0 does not work with filterless games due to security vulnerabilties. Please consider using Filtering or use ModRemote 2.7x");
-end
 
-if (script.Parent ~= game:GetService("ReplicatedStorage")) then
-	error("[ModRemote] Parent of Module should be ReplicatedStorage.");
-end
+-- Constants
+local client_Max_Wait_For_Remotes = 1
+local default_Client_Cache = 10
 
+-- Services
+local ReplicatedStorage	= game:GetService("ReplicatedStorage")
+local server		= game:GetService("NetworkServer")
```

Comment:
Should be `FindService("NetworkServer")` instead of `GetService("NetworkServer")` - will cause errors on the client because the client does not have a NetworkServer.